src/coreclr/jit/unwindamd64.cpp

```diff
@@ -399,7 +423,7 @@ void Compiler::unwindSaveRegWindows(regNumber reg, unsigned offset)
             code             = (UNWIND_CODE*)&func->unwindCodes[func->unwindCodeSlot -= sizeof(UNWIND_CODE)];
             code->UnwindOp   = (genIsValidFloatReg(reg)) ? UWOP_SAVE_XMM128_FAR : UWOP_SAVE_NONVOL_FAR;
         }
-        code->OpInfo          = (BYTE)reg;
+        code->OpInfo          = (BYTE)(genIsValidFloatReg(reg) ? reg - XMMBASE : reg);
```

Comment:
Can we add an assert that `genIsValidIntReg(reg)` for the else case, to help catch any future bugs or expansions in this space?

Reply:
We could even assert on the actual range of the computed value, since the unwind codes are well-defined and if they change, we want the assert to hit.

Reply:
let me do it as a follow up PR and merge this as is to unblock weekend runs.